### PR TITLE
Update helm version to 3.8.2 from 3.2.1

### PR DESCRIPTION
**Pull request by Szarny**
**What this PR does / why we need it**:

Update helm version to 3.8.2 from 3.2.1 to support future OCI Registry features, etc.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Update default helm version to 3.8.2
```

### CODE REVIEW

**Comment by nghialv** [comment]
Let's update the version of this docker image to v0.3.0
https://github.com/pipe-cd/pipecd/blob/master/tool/piped-base/DOCKER_BUILD

File: tool/piped-base/dockertest/check-helm.sh [hunk 1]
```diff
@@ -36,11 +36,11 @@ done
...
-versioncases["helm"]="version.BuildInfo{Version:\"v3.2.1\", GitCommit:\"fe51cd1e31e6a202cba7dead9552a6d418ded79a\", GitTreeState:\"clean\", GoVersion:\"go1.13.10\"}"
+versioncases["helm"]="version.BuildInfo{Version:\"v3.8.2\", GitCommit:\"6e3701edea09e5d55a8ca2aae03a68917630e91b\", GitTreeState:\"clean\", GoVersion:\"go1.17.5\"}"
 
 for h in "${!versioncases[@]}"
 do
-    got=$($h version --client)
+    got=$($h version --client --registry-config ${HELM_REGISTRY_CONFIG})
     want=${versioncases[$h]}
     if [[ ${got} == ${want} ]]; then
         echo -e "PASSED: Correct version for ${h}."
```
Comment (nghialv) [inline tool/piped-base/dockertest/check-helm.sh:46]: @Szarny We are on the way to getting rid of Kapetanios from PipeCD so you can remove all of those docker tests.

Comment (Szarny) [reply tool/piped-base/dockertest/check-helm.sh:46]: @nghialv Thank you for your confirmation. I checked that the correct version of Helm is installed on the local environment so I'll remove it.

**Review by nghialv — APPROVED** [review]
Here you go!

**Review by khanhtc1202 — APPROVED** [review]
🚀

**Pull request merged**